### PR TITLE
Add negation support for regex filters

### PR DESCRIPTION
**Pull request by kadaan**

### CODE REVIEW

File: resource/filter.go [hunk 1]
```diff
@@ -167,3 +177,24 @@ func (f Filter) matches(r *Resource) bool {
 	}
 	return false
 }
+
+func (f *StringFilter) matches(s string) (matched bool, err error) {
```
Comment (jckuester) [inline resource/filter.go]: I guess, we don't need named returned values here, so  `(bool, error)` is ok :+1:

File: resource/filter.go [hunk 2]
```diff
@@ -167,3 +177,24 @@ func (f Filter) matches(r *Resource) bool {
...
+	ok, err := regexp.MatchString(f.Pattern, s)
+	if err == nil && f.Negate {
```
Comment (jckuester) [inline resource/filter.go]: I'd prefer simple error checking constructs for readability:
``
```suggestion
	if err != nil {
	  return false, err
	}
	
	if f.Negate {
	  return !ok, nil
	}
	
	return ok, nil
	
```

**Comment by jckuester** [comment]
Looks good :+1: I think we should merge this next. Hope I get the build fixed soon. Can you add the `NOT` to the README?

File: resource/filter.go [hunk 3]
```diff
@@ -167,3 +177,33 @@ func (f Filter) matches(r *Resource) bool {
...
+	  return false, err
+	}
+	
+	if f.Negate {
+	  return !ok, nil
+	}
+	
+	return ok, nil
+	
+		ok = !ok
```
Comment (jckuester) [inline resource/filter.go]: Ah, there is some code left that my suggestion didn't replace. Also, can you please rebase? There are some conflicts.

File: resource/filter.go (hunk 1 above)
Comment (kadaan) [reply resource/filter.go]: Fixed

File: resource/filter.go (hunk 3 above)
Comment (kadaan) [reply resource/filter.go]: Fixed

**Comment by kadaan** [comment]
@jckuester Ready for review

**Comment by jckuester** [comment]
Still problems with the CI build (https://gitlab.com/jckuester/awsweeper/-/jobs/289751538): I set the docker image to `1.12` instead of `latest`, which helped with the previous problem, but now the `make generate` phase fails. 

Will try to reproduce that locally with by using a golang Docker container.

**Comment by jckuester** [comment]
Ran integration tests locally for now (-> all green) and released the newest changes: https://github.com/cloudetc/awsweeper/releases/tag/v0.3.3

Thanks :+1:

**Pull request merged**